src/plugins/kibana_react/public/code_editor/code_editor.tsx

```diff
@@ -154,6 +154,32 @@ export class CodeEditor extends React.Component<Props, {}> {
 
     this._editor = editor;
 
+    editor.addCommand(
+      monaco.KeyCode.Escape,
+      () => {
+        editor.trigger('Change tab focus mode', 'editor.action.toggleTabFocusMode');
+
+        const { tabFocusMode } = this._editor.getConfiguration();
+        const ariaLabel = this.props.options && this.props.options.ariaLabel;
+        const extraEditorClassName =
+          this.props.options && this.props.options.extraEditorClassName
+            ? this.props.options.extraEditorClassName
+            : '';
+
+        const ariaTabLabel = tabFocusMode
+          ? 'Tab moves focus. To toggle tab mode, press escape'
+          : 'Tab currently adds tab character. To change tab mode, press escape.';
+
+        editor.updateOptions({
+          extraEditorClassName: `${extraEditorClassName} ${
+            tabFocusMode ? 'monacoEditor--tabMode' : ''
+          }`,
+          ariaLabel: ariaLabel ? `${ariaLabel}. ${ariaTabLabel}` : ariaTabLabel,
```

Comment:
This might be some hassle to do, but could we move the `ariaTabLabel` from `ariaLabel` to [`ariaDescribedBy`](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Techniques/Using_the_aria-describedby_attribute)?

(`ariaDescribedBy` is a better place to put instructions and descriptions, whereas `ariaLabel` is for naming. Unfortunately, however, `ariaDescribedBy` takes an `id` of another element and reads that instead of allowing you to pass in text directly.)